api/buildings/serializers.py

```diff
@@ -1,9 +1,27 @@
 from rest_framework import serializers
 
-from .models import Building
+from .models import Building, Statistic
 
 
 class BuildingSerializer(serializers.ModelSerializer):
     class Meta:
         model = Building
         exclude = ()  # TODO: restrict to the necessary fields if needed
+
+
+class StatisticSerializer(serializers.ModelSerializer):
+    evaluated_buildings = serializers.SerializerMethodField(
+        "get_total_buildings"
+    )
+
+    def get_total_buildings(self, obj):
+        total_buildings = Building.objects.count()
+        return int(total_buildings)
```

Comment:
Would there be a way to run the `get_total_buildings` when adding new buildings and save that value in the database? I'm guessing `objects.count()` isn't the most intensive query though I wouldn't want to run it on a table with 2.5k entries (for now) each time someone opens the front page.

Reply:
Don't worry about that, django will cache that queryset.
Read this: [Caching and QuerySets](https://docs.djangoproject.com/en/3.1/topics/db/queries/#caching-and-querysets)
I tested with over 50000 buildings in DB and is very fast.
Also, we need to add pagination to DRF in settings otherwise will return all the buildings on the `api/v1/buildings` endpoint.
Try this:
```
docker-compose exec api bash
python manage.py shell
In [1]: from django.db import connection, reset_queries
In [2]: from buildings.models import Building
In [3]: total = Building.objects.count()
In [4]: total
Out[4]: 50006
In [5]: connection.queries
Out[5]: 
[{'sql': 'SELECT COUNT(*) AS "__count" FROM "buildings_building"',
  'time': '0.004'}]
In [6]: reset_queries()
In [7]: connection.queries
Out[7]: []
In [8]: total
Out[8]: 50006
```

Reply:
Alrighty, then. We'll go with it this way. Cheers!